source/game.screen.supermarket.production.bmx

```diff
@@ -1524,6 +1524,9 @@ Type TGUISelectCastWindow Extends TGUIProductionModalWindow
 			genderID = TVTPersonGender.GetAtIndex( genderFilterSelect.GetSelectedEntry().data.GetInt("genderIndex") )
 		EndIf
 		LoadPersons(jobID, genderID)
+		'sort again by simulating click on the previous sort type
+		sortType = sortType -1
+		onClickSortCastButton(null)
```

Comment:
das wirkt sehr "hacky" :dagger: 

Dann lieber die Sortierung vom "onClick" separieren und explizit dann sortieren koennen.

```
	Method onClickSortCastButton:Int(triggerEvent:TEventBase )
		sortType = sortType + 1
		If sortType > 3 Then sortType = 0
		If sortType = 0
			castSelectList.entries.sort(True, TGUICastSelectList.SortCastByName)
			sortCastButton.caption.SetSpriteName("gfx_datasheet_icon_az")
			sortCastTooltip.SetContent( StringHelper.UCFirst(GetLocale("NAME")) )
		Else If SortType = 1
			castSelectList.entries.sort(True, TGUICastSelectList.SortCastByJobXP)
			sortCastButton.caption.SetSpriteName("gfx_datasheet_icon_quality")
			sortCastTooltip.SetContent( StringHelper.UCFirst(GetLocale("CAST_JOB_EXPERIENCE")) )
		Else If SortType = 2
			castSelectList.entries.sort(True, TGUICastSelectList.SortCastByGenreXP)
			sortCastButton.caption.SetSpriteName("gfx_datasheet_icon_genreXP")
			sortCastTooltip.SetContent( StringHelper.UCFirst(GetLocale("CAST_GENRE_EXPERIENCE")) )
		Else If SortType = 3
			castSelectList.entries.sort(True, TGUICastSelectList.SortCastByFee)
			sortCastButton.caption.SetSpriteName("gfx_datasheet_icon_money")
			sortCastTooltip.SetContent( StringHelper.UCFirst(GetLocale("CAST_FEE")) )
		End If
		castSelectList.Update()
		castSelectList.UpdateLayout()
		Return True
	End Method
```

```
	Method onClickSortCastButton:Int(triggerEvent:TEventBase )
		sortType = sortType + 1
		If sortType > 3 Then sortType = 0
		SortCastList(sortType)

		Return True
	End Method
	
	Method SortCastList(sortType:Int = 0)            
		If sortType = 0
			castSelectList.entries.sort(True, TGUICastSelectList.SortCastByName)
			sortCastButton.caption.SetSpriteName("gfx_datasheet_icon_az")
			sortCastTooltip.SetContent( StringHelper.UCFirst(GetLocale("NAME")) )
		Else If SortType = 1
			castSelectList.entries.sort(True, TGUICastSelectList.SortCastByJobXP)
			sortCastButton.caption.SetSpriteName("gfx_datasheet_icon_quality")
			sortCastTooltip.SetContent( StringHelper.UCFirst(GetLocale("CAST_JOB_EXPERIENCE")) )
		Else If SortType = 2
			castSelectList.entries.sort(True, TGUICastSelectList.SortCastByGenreXP)
			sortCastButton.caption.SetSpriteName("gfx_datasheet_icon_genreXP")
			sortCastTooltip.SetContent( StringHelper.UCFirst(GetLocale("CAST_GENRE_EXPERIENCE")) )
		Else If SortType = 3
			castSelectList.entries.sort(True, TGUICastSelectList.SortCastByFee)
			sortCastButton.caption.SetSpriteName("gfx_datasheet_icon_money")
			sortCastTooltip.SetContent( StringHelper.UCFirst(GetLocale("CAST_FEE")) )
		End If
		castSelectList.Update()
		castSelectList.UpdateLayout()
	End Method
```

In obigem Code dann:
```
		'sort again
		SortCastList(sortType)
```

Reply:
Ich stimme zu. Der Sortiercode war vorher schon sehr hacky. Der Fix sollte minimalinvasiv sein ;-)